src/Components/WebAssembly/Build/src/Tasks/BlazorGetFileHash.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.Build.Tasks
+{
+    /// <summary>
+    /// Computes the checksum for a single file.
+    /// </summary>
+    public sealed class BlazorGetFileHash : Task
+    {
+        internal const string _defaultFileHashAlgorithm = "SHA256";
+        internal const string _hashEncodingBase64 = "base64";
+
+        /// <summary>
+        /// The files to be hashed.
+        /// </summary>
+        [Required]
+        public ITaskItem[] Files { get; set; }
+
+        /// <summary>
+        /// The algorithm. Allowed values: SHA256, SHA384, SHA512. Default = SHA256.
+        /// </summary>
+        public string Algorithm { get; set; } = _defaultFileHashAlgorithm;
+
+        /// <summary>
+        /// The metadata name where the hash is stored in each item. Defaults to "FileHash".
+        /// </summary>
+        public string MetadataName { get; set; } = "FileHash";
+
+        /// <summary>
+        /// The encoding to use for generated hashs. Defaults to "hex". Allowed values = "base64".
+        /// </summary>
+        public string HashEncoding { get; set; } = _hashEncodingBase64;
+
+        /// <summary>
+        /// The hash of the file. This is only set if there was one item group passed in.
+        /// </summary>
+        [Output]
+        public string Hash { get; set; }
+
+        /// <summary>
+        /// The input files with additional metadata set to include the file hash.
+        /// </summary>
+        [Output]
+        public ITaskItem[] Items { get; set; }
+
+        public override bool Execute()
+        {
+            if (!SupportsAlgorithm(Algorithm))
```

Comment:
Consider combining CreateAlgorithm and SupportsAlgorithms into one `Dictionary<string,Func<HashAlgorithm>>` so that only one place needs to be updated with new values

```C#
if (!_factories.TryGetValue(Algorithm, out var factory)) {
    Log.LogError("Unrecognized HashAlgorithm: {0}", Algorithm);
    return false;
}

...

Parallel.Foreach(Filles, file =>
{
    ... ComputeHash(factory, file);
...
});


...

private static readonly Dictionary<string, Func<HashAlgorithm>> factories = new HashAlgorithm(StringComparer.OrdinalIgnoreCase)
{
    { "SHA256", () => SHA256.Create() },
    { "SHA384", () => SHA384.Create() },
    { "SHA512", () => SHA512.Create() },
};
```